core/modules/comment/comment.module

```diff
@@ -1032,44 +1032,36 @@ function comment_build_content(Comment $comment, Node $node, $view_mode = 'full'
 function comment_links(Comment $comment, Node $node) {
   $links = array();
   if ($node->comment == COMMENT_NODE_OPEN) {
-    if (user_access('administer comments') && user_access('post comments')) {
-      $links['comment-delete'] = array(
-        'title' => t('delete'),
-        'href' => "comment/$comment->cid/delete",
-        'html' => TRUE,
-      );
-      $links['comment-edit'] = array(
-        'title' => t('edit'),
-        'href' => "comment/$comment->cid/edit",
-        'html' => TRUE,
-      );
-      $links['comment-reply'] = array(
-        'title' => t('reply'),
-        'href' => "comment/reply/$comment->nid/$comment->cid",
-        'html' => TRUE,
-      );
-      if ($comment->status == COMMENT_NOT_PUBLISHED) {
-        $links['comment-approve'] = array(
-          'title' => t('approve'),
-          'href' => "comment/$comment->cid/approve",
-          'html' => TRUE,
-          'query' => array('token' => backdrop_get_token("comment/$comment->cid/approve")),
+    if ($comment->access('delete') || $comment->access('edit') || Comment::createAccess() || $comment->access('approve')) {
```

Comment:
Minor, but we could tidy this up by skipping doing these checks twice. And instead if doing an if/else to show the `comment-forbidden` link, check `if (empty($links)) {` (since none of the other links will have been added).

Reply:
outdated.